iree/hal/cuda/cuda_headers.h

```diff
@@ -9,4 +9,18 @@
 
 #include "cuda.h"  // IWYU pragma: export
 
+#if IREE_ENABLE_CUPTI  // CUPTI is only used for tracing.
+
+/* Activity, callback, event and metric APIs */
+#include <cupti_activity.h>
+#include <cupti_callbacks.h>
+#include <cupti_events.h>
+#include <cupti_metrics.h>
+
+/* Runtime, driver, and nvtx function identifiers */
+#include <cupti_driver_cbid.h>
+#include <cupti_runtime_cbid.h>
+
+#endif  // IREE_TRACING_FEATURES
```

Comment:
```suggestion
#endif  // IREE_ENABLE_CUPTI
```

Reply:
Not quite, though maybe with some refactoring. The define check is typically

```c
#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
```

**note**: you also need this include for the tracing macros to be set up:

```c
#include "iree/base/tracing.h"
```

Reply:
Updated to `#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION`. Note that tracing.h was moved to the header file because I need to guard against additional includes.